lib/AppInfo/Application.php

```diff
@@ -67,8 +69,17 @@ private function registerServices() {
 				$c->query('DataSource')
 			);
 		});
+
+	}
+
+	private function registerHooks() {
+		// Activate data sources after user has successfully authenticated
+		$container = $this->getContainer();
+		$datasource = $container->query('DataSource');
+		\OCP\Util::connectHook('OC_App', 'loadedApps', $datasource, 'activateDataSources');
+		\OCP\Util::connectHook('OC_User', 'post_login', $datasource, 'activateDataSources');
```

Comment:
connect to hooks to be able to activate DataSources when we are sure user is authenticated. This will call activateDataSources() method, which will activate Event and Query Loggers, if not yet activated because of "debug mode" globaly.